tests/Benchmark.test.ts

```diff
@@ -51,111 +53,824 @@ const buildLSP6TestContext = async (
+            "transfer LYX from UP to 3x EOA with 256 bytes data",
+            receipt.gasUsed.toNumber().toString(),
+          ]);
+        });
+
+        it("Sending LYX from UP to UP with random data 256 bytes prefixed with 4 zeros", async () => {
+          const random256BytesData =
+            "0x00000000" +
+            ethers.utils.hexlify(ethers.utils.randomBytes(252)).substr(2);
+
+          const tx = await context.universalProfile
+            .connect(context.owner)
+            ["execute(uint256[],address[],uint256[],bytes[])"](
+              [
+                OPERATION_TYPES.CALL,
+                OPERATION_TYPES.CALL,
+                OPERATION_TYPES.CALL,
+              ],
+              [
+                universalProfile1.address,
+                universalProfile2.address,
+                universalProfile3.address,
+              ],
+              [
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+                ethers.utils.parseEther("0.1"),
+              ],
+              [random256BytesData, random256BytesData, random256BytesData]
+            );
+
+          const receipt = await tx.wait();
+
+          executeUP.push([
+            "transfer LYX from UP to 3x UP with 256 bytes data",
+            receipt.gasUsed.toNumber().toString(),
+          ]);
+        });
+      });
+
+      after(async () => {
+        UniversalProfileExecuteTable = getMarkdownTable({
+          table: {
+            head: ["`execute` scenarios - 👑 UP Owner", "⛽ Gas Usage"],
+            body: executeUP,
+          },
+          alignment: [Align.Left, Align.Center],
+        });
+      });
+    });
+
+    describe("setData", () => {
+      describe("setData Single", () => {
+        before(async () => {
+          context = await buildUniversalProfileContext(
+            ethers.utils.parseEther("50")
+          );
+        });
+        it("setting data of length 20 bytes", async () => {
+          let key = ethers.utils.keccak256(ethers.utils.toUtf8Bytes("My Key"));
+          let value = ethers.utils.hexlify(ethers.utils.randomBytes(20));
+
+          const tx = await context.universalProfile["setData(bytes32,bytes)"](
+            key,
+            value
+          );
+
+          const receipt = await tx.wait();
+
+          setDataUP.push([
+            "setting data of length 20 bytes",
+            receipt.gasUsed.toNumber().toString(),
+          ]);
+        });
+
+        it("setting data of length 60 bytes", async () => {
+          let key = ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("My Other Key")
+          );
+          let value = ethers.utils.hexlify(ethers.utils.randomBytes(60));
+
+          const tx = await context.universalProfile["setData(bytes32,bytes)"](
+            key,
+            value
+          );
+
+          const receipt = await tx.wait();
+
+          setDataUP.push([
+            "setting data of length 60 bytes",
```

Comment:
Maybe this would read better in the table?

```suggestion
            "set a 60 bytes long value",
```